### ISSUE TITLE
Out-of-Tree YAML and DTS support

### ISSUE THREAD

**Issue by jli157** [zephyrproject-rtos/zephyr#9198]
Is it possible to have yaml definitions and dts files in an out-of-tree board directory? 

```
myproject/
         boards/arm/myboard/myboard.dts
         dts/bindings/my_peripheral_dev.yaml
                             
```

This request could be part of the issue: #9172

**Comment by hakonfam** [comment]
Yes, use the `BOARD_ROOT` variable as specified in [here](http://docs.zephyrproject.org/application/application.html#custom-board-definition).

**Comment by jli157** [comment]
BOARD_ROOT doesn't support defining out-of-tree yaml files since the script `extract_dts_includes.py` only reads yaml files from the directory "${ZEPHYR_BASE}/dts/bindings" which is specified on the line 105 of the file `${ZEPHYR_BASE}/cmake/dts.cmake`.

**Comment by pizi-nordic** [comment]
FYI: @pdunaj

**Comment by pdunaj** [comment]
The same issue: https://github.com/zephyrproject-rtos/zephyr/issues/8849

**Comment by jli157** [comment]
Closed since the issue #8849 already covers the concerns this issue raises.
